packages/client/logic/clicks.ts

```diff
@@ -0,0 +1,109 @@
+import { sum } from '@antfu/utils'
+import type { ClicksContext } from '@slidev/types'
+import type { Ref } from 'vue'
+import { ref, shallowReactive } from 'vue'
+import type { RouteRecordRaw } from 'vue-router'
+import { currentRoute, isPrintMode, isPrintWithClicks, queryClicks, routeForceRefresh } from './nav'
+import { safeParseNumber } from './utils'
+
+export function usePrimaryClicks(route: RouteRecordRaw | undefined): ClicksContext {
+  if (route?.meta?.__clicksContext)
+    return route.meta.__clicksContext
+  const thisPath = +(route?.path ?? 99999)
+
+  const flow = new Map()
+  const map = shallowReactive(new Map())
+
+  const context: ClicksContext = {
+    get disabled() {
+      return isPrintMode.value && !isPrintWithClicks.value
+    },
+    get current() {
+      const currentPath = +(currentRoute.value?.path ?? 99999)
+      if (currentPath === thisPath)
+        return queryClicks.value
+      else if (currentPath > thisPath)
+        return 99999
+      else
+        return 0
+    },
+    flow,
+    map,
+    register(el, resolved) {
+      flow.set(el, resolved.flowSize)
+      map.set(el, resolved)
+    },
+    unregister(el) {
+      flow.delete(el)
+      map.delete(el)
+    },
+    get flowSum() {
+      // eslint-disable-next-line no-unused-expressions
+      routeForceRefresh.value
+      return sum(...flow.values())
+    },
+    get total() {
+      // eslint-disable-next-line no-unused-expressions
+      routeForceRefresh.value
+      return route?.meta?.clicks ?? (map?.size
+        ? Math.max(...[...map.values()].map(v => v.max))
+        : 0)
+    },
+  }
+  route?.meta && (route.meta.__clicksContext = context)
+  return context
+}
+
+export function useFixedClicks(route: RouteRecordRaw | undefined, currentInit = 0): [Ref<number>, ClicksContext] {
+  const current = ref(currentInit)
+  const flow = new Map()
+  const map = shallowReactive(new Map())
+
+  return [current, {
+    get disabled() {
+      return isPrintMode.value && !isPrintWithClicks.value
+    },
+    get current() {
+      return current.value
+    },
+    set current(_v) {
+    },
+    flow,
+    map,
+    register(el, resolved) {
+      flow.set(el, resolved.flowSize)
+      map.set(el, resolved)
+    },
+    unregister(el) {
+      flow.delete(el)
+      map.delete(el)
+    },
+    get flowSum() {
+      // eslint-disable-next-line no-unused-expressions
+      routeForceRefresh.value
+      return sum(...flow.values())
```

Comment:
The `flow` map seems to be only used here. Maybe we could use the `[...map.values()].map(i=>i.flowSize)` directly and have a single source of truth?

Reply:
I've just combined `flow` into `map`, but the e2e tests failed because of infinite recursive. This because (实在抱歉我没法用英文准确地表达下面这段话) `flow`是在运行时由v-click指令和组件的`onMounted`钩子一边添加新的元素，一边读取最新的累计数据的，**不能是响应式的，不然就会陷入死循环**；而`map`的数据是在所有点击元素都已经解析好之后才能用的，**应当是响应式的**。我暂时想不到把flow合并到map的办法😭

Reply:
I will give it a try after you finish it, you could leave the two maps there for now